src/main/java/com/derongan/minecraft/deeperworld/DeeperWorld.java

```diff
@@ -10,22 +11,19 @@
 
 public final class DeeperWorld extends JavaPlugin {
     private WorldManagerImpl worldManager;
-    private MovementListener movementListener;
-    private SectionSyncListener sectionSyncListener;
 
     @Override
     public void onEnable() {
         createConfig();
 
+        //TODO convert PlayerManager, WorldManager, and MovementListener to objects
         PlayerManager playerManager = new PlayerManager();
 
         worldManager = new WorldManagerImpl(getConfig());
         this.getServer().getServicesManager().register(WorldManager.class, worldManager, this, ServicePriority.Lowest);
 
-        movementListener = new MovementListener(playerManager);
-        sectionSyncListener = new SectionSyncListener();
-        this.getServer().getPluginManager().registerEvents(movementListener, this);
-        this.getServer().getPluginManager().registerEvents(sectionSyncListener, this);
+        this.getServer().getPluginManager().registerEvents(new MovementListener(playerManager), this);
+        this.getServer().getPluginManager().registerEvents(SectionSyncListener.INSTANCE, this);
```

Comment:
why singleton

Reply:
Makes it easier to access stuff inside, and I thought it was appropriate for listeners since there should only ever be one instance. I think it makes sense for the managers too, but I can see counter arguments for it. If you know good reasons not to do things this way, it's not a big deal and I can revert.